Feature/#386 rtp command - second try

opmis opened the pull request:
Closes #184 

# Changes proposed in this pull request:
- added /mc home
- cleaned up permission checks to be simplified
-

Review please

opmis committed:
Colony TP Command

Random TP players to location away from other colonies

opmis committed:
ColoyTPCommand update

Added in a 4 try for the TP then requests player to try again if it does not find a good spot

Raycoms committed:
Fix problem with world not loading.

Better use less global variables and distribute the variables to the methods and use the return values.
We can't be sure if the sender is a player always. 
And if we instantiate that in the variables already probably causes the bug you had.

opmis committed:
Colony TP Command 

Y not reassigning

opmis committed:
Colony TP Command

Seems to be working good

opmis committed:
Colony TP Command

add the permission to the config file to allow/deny players to use of it

opmis committed:
Colony RTP Command

set up config file to handle the ability to allow players to use the command or not

opmis committed:
Added perms for 8 commands to configs

Added the perms to the commands we discussed and added a configuration for the upper bounds for the TP command

opmis committed:
Commands with Configs

Simplified code to check the config file to see with players can use commands

opmis committed:
Commands Clean up

Still getting a couple errors when using commands

opmis committed:
Commands

Fix CitizenInfo  pushing for safety as I fix the others

opmis committed:
Commands

All working still need testing in a Multiplayer environment.  addOfficer still has issues.

Raycoms committed:
Hotfix/some bugs (#545)

* Fix game crashing bug and let players attack mobs

* Check if file exists

* remove unused imports

* change order of if

* Coltons remark

Raycoms committed:
clean up commands #1

opmis committed:
Commnads

mainly just clean up

Raycoms committed:
More commands clean up-

Raycoms committed:
Clean up the colonyTPCommand

Raycoms committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/#386-Rtp-command'

Conflicts:
	src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java
	src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java
	src/main/java/com/minecolonies/coremod/commands/DeleteColonyCommand.java
	src/main/java/com/minecolonies/coremod/commands/ShowColonyInfoCommand.java

Raycoms committed:
Fix up some minor issues.

Raycoms committed:
improve save position check

Raycoms committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

Conflicts:
	src/main/java/com/minecolonies/coremod/commands/CitizenInfoCommand.java
	src/main/java/com/minecolonies/coremod/commands/ListCitizensCommand.java
	src/main/java/com/minecolonies/coremod/commands/ShowColonyInfoCommand.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

Raycoms committed:
fix javadoc

opmis committed:
Commands

All polished and spit shined...

xavierh committed:
Fix upgrade button label for town hall (#543)

* Fix upgrade button label for town hall

* "Upgrade Building" button label fix

* upgrade should be enabled

* revert back to the previous translation string

Raycoms committed:
Merged branch develop into feature/#386-Rtp-command

Raycoms committed:
Recall shouldn't return anything without citizen.

Raycoms committed:
Add refresh and change owner commands.

Raycoms committed:
Fix compile error

opmis committed:
/#386-Rtp-command:

Made suggested changes to CTP

opmis committed:
Commands

added OP over ride of permissions to Change and restore Colony Owner

Raycoms committed:
more fixes

opmis committed:
Commands

Finished moving some of the methods to utilis

opmis committed:
Merged branch feature/#386-Rtp-command into feature/#386-Rtp-command

opmis committed:
Commands

CTP Works again.  A little cleanup

Raycoms committed:
Improve commands. (fix rtp and add to  colonyOwner)

Raycoms committed:
fix up owner command even more

Raycoms committed:
Fix command

Raycoms committed:
Fix up owner command and permissions.

Raycoms committed:
Merge branch 'develop'

Conflicts:
	src/main/java/com/minecolonies/coremod/configuration/Configurations.java

Raycoms committed:
merge the repair build buttons.

Raycoms committed:
fix up merge problems.

Raycoms committed:
more merge fixes

Raycoms committed:
fix annotations.

Raycoms committed:
Resolved problems marvin mentioned

Raycoms committed:
Resolve sonar issues.

Raycoms committed:
Add missing parameter

Raycoms committed:
rename random teleport command

Raycoms committed:
Fix bug with incorrect pickaxe delivery

Raycoms committed:
fix up random teleport

Raycoms committed:
didn't need that check

opmis committed:
Fixin Delete and refresh

Delete still needs an override to check for Owner Only

opmis committed:
Merged branch feature/#386-Rtp-command into feature/#386-Rtp-command

Raycoms committed:
Fix up commands and imports

opmis committed:
mc Home command

TPs player to the center of thier colony

opmis committed:
command configs

fix the configuration calls to not be finals

cltnschlosser commented:
```
/home/travis/build/Minecolonies/minecolonies/build/sources/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java:155: warning: [cast] redundant cast to JobBuilder
            ((JobBuilder) job).setStructure(null);
             ^
/home/travis/build/Minecolonies/minecolonies/build/sources/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java:156: warning: [cast] redundant cast to JobBuilder
            ((JobBuilder) job).setWorkOrder(null);
             ^
2 warnings
:processResources UP-TO-DATE
:classes
:javadoc/home/travis/build/Minecolonies/minecolonies/src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java:275: error: @param name not found
     * @param playerName the player to set.
              ^
/home/travis/build/Minecolonies/minecolonies/src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java:278: warning: no @param for player
    public boolean setOwner(final EntityPlayer player)
                   ^
1 error
1 warning
:javadoc FAILED
```

marvin-bitterlich commented:
This pr is missing a description

Raycoms committed:
Fix up merge conflicts

Raycoms committed:
last merge conflicts

opmis committed:
/#386-Rtp-command:

code is looking good.  Corrected a PERNUM issue with the backup command and set i up to use the proper configuration permmissions.  Also corrected in code documention for ChangeColonyOwner

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:
```diff
@@ -17,14 +17,13 @@
+        CITIZENINFO, COLONYTP, DELETECOLONY, KILLCITIZENS, LISTCITIZENS, RESPAWNCITIZENS, SHOWCOLONYINFO, ADDOFFICER, CHANGE_COLONY_OWNER, REFRESH_COLONY, HOMETP, MC_BACKUP
     }
 
     /**
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

Raycoms replied:
Yeah you can't make it **

opmis replied:
No clue what this is talking about

marvin-bitterlich replied:
Just remove the **

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,126 @@
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException
+    {
+
+        EntityPlayer playerToTeleport = null;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "playerToTeleport". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the declaration of "playerToTeleport" closer to the code that uses it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1941)

MinecoloniesSonar commented:
SonarQube analysis reported 11 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L115](https://github.com/Minecolonies/minecolonies/blob/29987d55d79749d71a092c6b8d2446c3fe41fd4d/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L115): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,126 @@
+    {
+
+        EntityPlayer playerToTeleport = null;
+        IColony colony = null;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "colony". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the declaration of "colony" closer to the code that uses it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1941)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,126 @@
+
+        EntityPlayer playerToTeleport = null;
+        IColony colony = null;
+        int colonyId = 0;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "colonyId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the declaration of "colonyId" closer to the code that uses it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1941)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,126 @@
+            }
+
+
+            if (colony.equals(null))
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2583)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Remove this call to "equals"; comparisons against null always return false; consider using '== null' to check for nullity. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2159)

Raycoms replied:
yeah please do this.

Raycoms replied:
this still needs work

Raycoms replied:
Still please

opmis replied:
OK FIXED

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,126 @@
+     * @param colID           the senders colony ID.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private static void teleportPlayer(final EntityPlayer playerToTeleport,int colID) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

opmis committed:
Sonar Clean up

Cleaned up issues detected by Sonar

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/ChangeColonyOwnerCommand.java:
```diff
@@ -62,37 +62,25 @@ public void execute(@NotNull final MinecraftServer server, @NotNull final IComma
                 return;
             }
-
+            EntityPlayer player = sender.getEntityWorld().getPlayerEntityByName(playerName);
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'player' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

MinecoloniesSonar commented:
SonarQube analysis reported 5 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L115](https://github.com/Minecolonies/minecolonies/blob/ffb02a6d7789409f137231c1f7bccc3a8757a6da/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L115): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

opmis committed:
fix Perm issue

Fixed OP permission issue with Backup and a lil more clean up

MinecoloniesSonar commented:
SonarQube analysis reported 4 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L115](https://github.com/Minecolonies/minecolonies/blob/7f07e6fa607e46c94e069d151efac99353dba615/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L115): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Raycoms commented on src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:
```diff
@@ -150,12 +153,12 @@ public boolean canCommandSenderUseCommand(Commands theCommand)
             requiredOpLevel = 1;
         }
-        if (PERMNUM > 4)
+        if (Configurations.opLevelForServer > 4)
```
I just remembered, does MathUtils.clamp(Value,1,4) not clamp the value between 1 and 4?

opmis replied:
I did this in the event that the number the user puts into the config is out of range, it will auto adjust it so as not to throw an error

Raycoms replied:
Yeah I know but I think MathUtils.clamp(value,1,4) will keep it also in that range but with way less code.

Raycoms replied:
MathHelper.clamp_int()

opmis replied:
oh that thing  didnt know it existed :)  thanks

opmis replied:
Fixed.  My code did look sexier though...

Raycoms commented on src/main/java/com/minecolonies/coremod/commands/BackupCommand.java:
```diff
@@ -35,11 +37,8 @@ public BackupCommand(@NotNull final String... parents)
     {
-        if (!sender.canCommandSenderUseCommand(PERMNUM, "minecolonies backup"))
-        {
-            sender.addChatMessage(new TextComponentString(NO_PERMISSION_MESSAGE));
```
I think we still need the "NO_PERMISSION_MESSAGE" feedback if the player isn't opped.

opmis replied:
my bad thought that got called out in the Op check.  FIXED

Raycoms commented on src/main/java/com/minecolonies/coremod/commands/ChangeColonyOwnerCommand.java:
```diff
@@ -62,37 +62,25 @@ public void execute(@NotNull final MinecraftServer server, @NotNull final IComma
             }
-
             final EntityPlayer player = sender.getEntityWorld().getPlayerEntityByName(playerName);
-            if(player == null)
```
we don't need to check for the colony and player null anymore?

opmis replied:
confused.  We do check for them separately

Raycoms replied:
the check got removed with this pr.

opmis replied:
I still have both.

opmis replied:
and its still sexy

Raycoms replied:
Ah they're way down there

Raycoms commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,126 @@
+    @Override
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException
+    {
+
```
Please correctly format this code. It seems a bit off.

opmis replied:
I don't even know what that means.......

Raycoms replied:
ctrl - alt - l

opmis replied:
every time I do it moves the curly bracket up to the line and then sonar freaks out because sonar is egotistical and thinks everything should be his way....

Raycoms commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,126 @@
+                sender.getCommandSenderEntity().addChatMessage(new TextComponentString(CANT_FIND_COLONY));
+                return;
+            }
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("There's no place like home, there's no place like home..."));
```
please take a look how I avoided too many teleport requests by a player in the random teleport command and implement it here as well.

opmis replied:
ok I am not seeing what you are talking about in the RTP.  I didnt think it would be an issue as this is an absolute position not one that we have to do a lot of things to generate

Raycoms replied:
It's more about to avoid players spamming the command or abusing it to hop between two positions.

opmis replied:
ok I am not seeing what you are talking about in the RTP

Raycoms replied:
In the RTP I am checking if the player can execute this command. I think its MinecoloniesCommand.canPlayerUse or similar. We need this check here as well before teleporting.

opmis replied:
oh I seen that and was thinking that was something else OK ill look at that

opmis replied:
ok this is fixed, but..... with the RTP dont we want to check the timer right off the bat?  right now it is doing all the calculations and the checking the timer, I would think would should check timer and if OK then do the calculations.

Raycoms replied:
No, because if the rtp command is unable to teleport the players counter shouldn't be increased.
And checking the availability increases the counter already.

Raycoms commented on src/main/java/com/minecolonies/coremod/commands/RandomTeleportCommand.java:
```diff
@@ -123,24 +119,16 @@ private static void teleportPlayer(final ICommandSender sender, final EntityPlay
-            /*If no position found*/
-            if (groundPosition == null)
+            /* Check for a close by colony*/
+            if (ColonyManager.getColony(sender.getEntityWorld(), new BlockPos(x, STARTING_Y, z)) != null || (groundPosition == null))
```
why are there brackets around the ground position check?

opmis replied:
I do not see this in my code but the brackets do look sexy

opmis committed:
more Command fixes and cleanup

Home TP now has a cool down, fixed the backup perms issue.  Made my code look sexy...

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    @Override
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

Raycoms replied:
Yes please

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,136 @@
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("Please wait at least " + Configurations.teleportBuffer + " seconds to teleport again"));
+        }
+        return;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Avoid unnecessary return statements [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUnnecessaryReturn)

Raycoms replied:
Yes please

MinecoloniesSonar commented:
SonarQube analysis reported 6 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/3dc8889ec7e91e71858db0d5a07589888e16c286/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

cltnschlosser commented on src/main/resources/assets/minecolonies/gui/windowhutfisherman.xml:
```diff
@@ -1,7 +1,7 @@
 <window xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
This rename to lowercase will break the window.  In code we expect windowHutFisherman.xml

Raycoms commented:
Ah, my fault. It's not a rename it's a conflict with some 1.11 merge. I'll
fix that.

2017-03-08 12:11 GMT-03:00 Colton Schlosser <notifications@github.com>:

> *@cltnschlosser* commented on this pull request.
> ------------------------------
>
> In src/main/resources/assets/minecolonies/gui/windowhutfisherman.xml
> <https://github.com/Minecolonies/minecolonies/pull/671#discussion_r104937759>
> :
>
> > @@ -1,7 +1,7 @@
>  <window xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
>
> This rename to lowercase will break the window. In code we expect
> windowHutFisherman.xml
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Minecolonies/minecolonies/pull/671#pullrequestreview-25794299>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGI9y7Am5KkiZZhwcxd4JDyLBrH-XpiHks5rjsUbgaJpZM4MS8ph>
> .
>

Raycoms committed:
rename to correct name for 1.10

MinecoloniesSonar commented:
SonarQube analysis reported 6 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/9a73750039bec7b0d18eadeb3562e0f0d2e1061b/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Raycoms commented:
already done in other pr
